packages/design-system/src/components/input/numericInput.tsx

```diff
@@ -88,14 +91,29 @@ const NumericInput = forwardRef(function NumericInput(
     [handleBlur]
   );
 
+  // Holds filteredValue that may have leading padZero when required.
+  let paddedValue = '';
+  if (max !== undefined) {
+    const maxPad = max !== undefined ? String(max).length : 0;
```

Comment:
```suggestion
    const maxPad = String(max).length;
```

We already checked `max !== undefined` above.

Reply:
Absolutely! I missed it. I am updating accordingly.